presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestRFC2822JsonDateTimeFormatter.java

```diff
@@ -0,0 +1,117 @@
+    {
+        String formattedStr = formatType.getFormatter(Optional.empty()).formatTime(value, precision);
+        assertEquals(RFC_FORMATTER.parse(formattedStr).getLong(NANO_OF_DAY), getNanosOfDay(scaleEpochPicosToNanos(value.getPicos())));
+    }
+
+    private void testTimeWithTZ(SqlTimeWithTimeZone value, FormatType formatType)
+    {
+        String formattedStr = formatType.getFormatter(Optional.empty()).formatTimeWithZone(value);
+        TemporalAccessor parseResult = RFC_FORMATTER.parse(formattedStr);
+        assertEquals(parseResult.get(MILLI_OF_DAY), getMillisOfDay(value.getMillisUtc()));
+        assertEquals(getTimeZoneKey(ZoneId.from(parseResult).getId()), UTC_KEY);
+    }
+
+    private void testTimestamp(SqlTimestamp value, FormatType formatType)
+    {
+        String formattedStr = formatType.getFormatter(Optional.empty()).formatTimestamp(value);
+        TemporalAccessor parseResult = RFC_FORMATTER.withZone(ZoneId.of("UTC")).parse(formattedStr);
+        assertEquals(DAYS.toMillis(parseResult.getLong(EPOCH_DAY)) + parseResult.getLong(MILLI_OF_DAY), value.getMillis());
+    }
+
+    private void testTimestampWithTZ(SqlTimestampWithTimeZone value, FormatType formatType)
+    {
+        String formattedStr = formatType.getFormatter(Optional.empty()).formatTimestampWithZone(value);
+
+        TemporalAccessor parseResult = RFC_FORMATTER.parse(formattedStr);
+        long actualEpochMillis = (parseResult.getLong(INSTANT_SECONDS) + parseResult.getLong(OFFSET_SECONDS)) * 1000 + parseResult.getLong(MILLI_OF_SECOND);
+
+        assertEquals(actualEpochMillis, value.getEpochMillis());
+        assertEquals(getTimeZoneKey(ZoneId.from(parseResult).getId()), UTC_KEY);
+    }
+
+    @Test
+    public void testRFC2822DateTimeFunctions()
+    {
+        FormatType formatType = FormatType.RFC2822;
+
+        testDate(new SqlDate(389356), formatType);
+        testDate(new SqlDate(0), formatType);
+        testDate(new SqlDate(Integer.MAX_VALUE), formatType);
+
+        testTime(SqlTime.newInstance(3, 75303000000000000L), formatType, 3);
+        testTime(SqlTime.newInstance(3, 0), formatType, 3);
+        testTime(SqlTime.newInstance(3, 86399000000000000L), formatType, 3);
+
+        testTimeWithTZ(new SqlTimeWithTimeZone(1594846503000L, TimeZoneKey.getTimeZoneKeyForOffset(-240)), formatType);
+        testTimeWithTZ(new SqlTimeWithTimeZone(1594846503000L, UTC_KEY), formatType);
+        testTimeWithTZ(new SqlTimeWithTimeZone(1594846503000L, TimeZoneKey.getTimeZoneKeyForOffset(-360)), formatType);
+        testTimeWithTZ(new SqlTimeWithTimeZone(9223372036000L, TimeZoneKey.getTimeZoneKeyForOffset(720)), formatType);
+        testTimeWithTZ(new SqlTimeWithTimeZone(0, UTC_KEY), formatType);
+
+        testTimestamp(SqlTimestamp.fromMillis(3, 1594846503000L), formatType);
+        testTimestamp(SqlTimestamp.fromMillis(3, 0), formatType);
+        testTimestamp(SqlTimestamp.fromMillis(3, 9223372036000L), formatType);
```

Comment:
Using these long values to construct timestamps makes it difficult to reason about the tests.  `DateTimeTestingUtils` has some utilities for constructing `SqlTimestamp` in a more readable form.

Some edge cases are better expressed with millis values, but you could use something like `NANOSECONDS.toMillis(Long.MAX_VALUE)` instead of `9223372036000L`.

As an aside, `SqlTimestamp` can represent a much wider range, so that's not an edge case.  It would also make sense to have some dates before the epoch.